include/silo/config/preprocessing_config.h

```diff
@@ -7,59 +7,52 @@
 #include <string_view>
 #include <unordered_map>
 
+#include <Poco/Util/OptionSet.h>
 #include <fmt/format.h>
 
-#include "silo/config/util/abstract_config_source.h"
+#include "config/config_interface.h"
+#include "config/source/yaml_file.h"
+#include "silo/config/config_defaults.h"
 
 namespace silo::config {
 
-const AbstractConfigSource::Option INPUT_DIRECTORY_OPTION{{"inputDirectory"}};
-const AbstractConfigSource::Option OUTPUT_DIRECTORY_OPTION = {{"outputDirectory"}};
-const AbstractConfigSource::Option INTERMEDIATE_RESULTS_DIRECTORY_OPTION = {
-   {"intermediateResultsDirectory"}
-};
-const AbstractConfigSource::Option PREPROCESSING_DATABASE_LOCATION_OPTION = {
-   {"preprocessingDatabaseLocation"}
-};
-const AbstractConfigSource::Option DUCKDB_MEMORY_LIMIT_OPTION = {{"duckdbMemoryLimitInG"}};
-const AbstractConfigSource::Option LINEAGE_DEFINITIONS_FILENAME_OPTION = {
-   {"lineageDefinitionsFilename"}
-};
-const AbstractConfigSource::Option NDJSON_INPUT_FILENAME_OPTION = {{"ndjsonInputFilename"}};
-const AbstractConfigSource::Option REFERENCE_GENOME_FILENAME_OPTION = {{"referenceGenomeFilename"}};
-
-const std::string DEFAULT_OUTPUT_DIRECTORY = "./output/";
-
 class PreprocessingConfig {
```

Comment:
There are no pure getters any more, the methods that remain are building return values. But, fair enough, those fields that are processed by methods should perhaps be private. @Taepper ?

Reply:
In theory, I agree with you. Two things to consider:
- I want to get rid of more of the "implicit relative paths logic" in the future. Then we really just use the config values as structs.
This by itself would still lead me to say: then have them private for now, because we might forget about it, but:
- The nlohmann definitions for debug printing are a lot more complicated when we have private members..

So I would advocate for leaving it as is for now?

Reply:
Fine by me

Reply:
I actually did manage to solve the above points. Feel free to check out the new code